### PR TITLE
Mon 35539 engine configuration manager telegraf

### PR DESCRIPTION
**Pull request by jean-christophe81**
REFS:MON-35539

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [X] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/scripts/collect-test-robot.sh [hunk 1]
```diff
@@ -62,6 +62,9 @@ echo '/tmp/core.%p' > /proc/sys/kernel/core_pattern
 #remove git dubious ownership
 /usr/bin/git config --global --add safe.directory $PWD
 
+echo "###### git clone opentelemetry-proto  #######"
+git clone https://github.com/open-telemetry/opentelemetry-proto.git opentelemetry-proto
```
Comment (kduret) [inline .github/scripts/collect-test-robot.sh:66]: you should use shallow clone to download only what you want and faster : `--depth=1 --single-branch --branch=<branch>`

File: broker/core/src/main.cc [hunk 2]
```diff
@@ -62,7 +63,9 @@ using namespace com::centreon::exceptions;
 
 std::shared_ptr<asio::io_context> g_io_context =
     std::make_shared<asio::io_context>();
-bool g_io_context_started = false;
+
+std::unique_ptr<com::centreon::common::pool>
+    com::centreon::common::pool::_instance;
```
Comment (bouda1) [inline broker/core/src/main.cc:69]: Would be better in pool.cc

File: engine/src/main.cc [hunk 3]
```diff
@@ -70,7 +72,9 @@ using namespace com::centreon::engine;
 
 std::shared_ptr<asio::io_context> g_io_context(
     std::make_shared<asio::io_context>());
-bool g_io_context_started = false;
+
+std::unique_ptr<com::centreon::common::pool>
+    com::centreon::common::pool::_instance;
```
Comment (bouda1) [inline engine/src/main.cc:77]: would be better in pool.cc

File: broker/core/src/main.cc (hunk 2 above)
Comment (jean-christophe81) [reply broker/core/src/main.cc:69]: tried and I have noticed duplication of symbols

File: engine/modules/opentelemetry/src/telegraf/nagios_converter.cc [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    res.output = state_str[res.exit_code];
+  }
+  res.output.push_back('|');
+  for (const auto& perf : perfs) {
+    if (perf.second.val) {
+      res.output += perf.first;
+      res.output.push_back('=');
+      absl::StrAppend(&res.output, *perf.second.val);
+      res.output += perf.second.unit;
+      res.output.push_back(';');
```
Comment (bouda1) [inline engine/modules/opentelemetry/src/telegraf/nagios_converter.cc:237]: No way to do something like
```
res.output += fmt::format("{}={}{};", perf.first, *perf.second.val, perf.second.unit);
```
Or with the absl functions....

Comment (bouda1) [reply engine/modules/opentelemetry/src/telegraf/nagios_converter.cc:237]: And maybe we can improve a little more with min, max, and thresholds.

File: tests/resources/opentelemetry/telegraf.conf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+   service_address = "127.0.0.1:4317"
+
+
+[[inputs.exec]]
+  name_override = "otel_check_icmp_host_1"
+  commands = ["/usr/lib/nagios/plugins/check_icmp 127.0.0.10"]
+  data_format = "nagios"
+  [inputs.exec.tags]
+    host = "host_1"
+    service = ""
```
Comment (bouda1) [inline tests/resources/opentelemetry/telegraf.conf:17]: Are these tabulations correct?
On line 8, I have an impression of too many spaces, am I wrong?

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+ *                   ]
+ *               }
+ *           ]
+ *       }
+ *   },
+ * @endcode
+ *
+ *
+ */
+class otl_nagios_converter : public otl_converter {
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh:94]: nagios_converter yet renamed in the next pr
otl_check_result_builder in the PR after

File: common/http/test/http_server_test.cc [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+                     cond.notify_one();
+                 });
+  }
+  std::unique_lock l(cond_m);
+  cond.wait(l);
+
+  client->shutdown();
+}
+
+TEST_P(http_server_test, only_connect_to_timeout) {
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply common/http/test/http_server_test.cc:375]: a client connects to server without send any request. It test server receive timeout

File: engine/src/main.cc [hunk 8]
```diff
@@ -1,22 +1,19 @@
...
- * General Public License for more details.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  *
- * You should have received a copy of the GNU General Public License
- * along with Centreon Engine. If not, see
- * <http://www.gnu.org/licenses/>.
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/src/main.cc:19]: done

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+** For more information : contact@centreon.com
+*/
+
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh:18]: done

File: engine/modules/opentelemetry/src/telegraf/nagios_converter.cc (hunk 4 above)
Comment (jean-christophe81) [reply engine/modules/opentelemetry/src/telegraf/nagios_converter.cc:237]: done

File: tests/resources/opentelemetry/telegraf.conf (hunk 5 above)
Comment (jean-christophe81) [reply tests/resources/opentelemetry/telegraf.conf:17]: done

**Review by bouda1 — APPROVED** [review]

**Pull request merged**